.pre-commit-config-cpp.yaml

```
@@ -0,0 +1,26 @@
+# This file contains pre-commit hooks for checking the C++ code. Since these
+# hooks require tools that may not be trivial to install (unlike the Python
+# hooks like flake8), we keep these in a separate file to be applied manually.
+repos:
+  - repo: local
+    hooks:
+      - id: clang-format
+        name: clang-format
+        entry: clang-format -i
+        language: system
+        types: [c++]
```

We should run clang-format on `.cu`, `.cuh`, and `.inc` files as well as standard C++ file types.

---

This suggestion won't catch `.cuh` via `types` yet -- I opened https://github.com/pre-commit/identify/pull/195 to address that.
```suggestion
        types_or: [c++, cuda, inc]
```